File: app/src/main/java/com/example/emptyactivity/MyApp.java
package com.example.emptyactivity;

import android.app.Application;

public class MyApp extends Application {
    /* Always be able to access the module ("static") */
    public static AppModule appModule;
    /* Called only once at beginning of application's lifetime */
    @Override
    public void onCreate() {
        super.onCreate();
        appModule = new AppModule(this);
    }
}

File: app/src/main/java/com/example/mutsu/MyApp.kt
package com.example.mutsu

import android.app.Application
import com.google.firebase.auth.ktx.auth
import com.google.firebase.ktx.Firebase

/*
 * The code and logic in this file was copied/modified from Talib's notes and
 * github repository code found here:
 * https://github.com/tshussain/KotlinWithCompose
 */

class MyApp: Application() {
    companion object {
        lateinit var appModule: AppModule
    }

    override fun onCreate() {
        super.onCreate()
        appModule = AppModule(this, Firebase.auth)
    }
}